File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-12.py
class Solution:
    def mergeTriplets(self, A: List[List[int]], B: List[int]) -> bool:
        if len(A) == 1:
            return A[0] == B

        for i in range(3):
            found = False
            minVal = float("inf")
            for j in range(len(A)):
                minVal = min(minVal, A[j][i])
                if B[i] == A[j][i]:
                    found = True
                
            if not found:
                return False

            if B[i] == minVal:
                return False

        return True

File: Data Structures & Algorithms/max-water-container/submission-12.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int mx = 0;
        int n = heights.size();
        int l = 0, r = 0;
        while (r < n) {
            r++;
            mx = max(mx, min(heights[l], heights[r]) * (r-l));
            if (heights[r] > heights[l]) {
                l = r;
            }
        }
        return mx;
    }
};